File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        // Algoritmo Guloso:
        // Verifica de trás para frente
        
        int meta = nums.size() - 1;
        for (int i = nums.size() - 1; i >= 0; i--){
               if (i + nums[i] >= meta){
                meta = i;
               }
        }
        if (meta == 0){
            return true;
        } else {
            return false;
        }
    }
};

File: 0055-jump-game/0055-jump-game.py
class Solution(object):
    def canJump(self, nums):
        meta = len (nums) - 1

        for i in range (len(nums) - 1, - 1, -1):
            if i + nums[i] >= meta:
                meta = i
        
        return meta == 0
            
        